platform-dependent-components/problem-solver/cxx/sections-module/agent/get_subdivision_agent.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * This source file is part of an OSTIS project. For the latest info, see http://ostis.net
+ * Distributed under the MIT License
+ * (See accompanying file COPYING.MIT or copy at http://opensource.org/licenses/MIT)
+ */
+
+#include "sc-agents-common/utils/AgentUtils.hpp"
+#include "sc-agents-common/utils/CommonUtils.hpp"
+#include "sc-agents-common/utils/IteratorUtils.hpp"
+#include "sc-agents-common/utils/GenerationUtils.hpp"
+#include "sc-core/sc-store/sc-container/sc_struct_node.h"
+#include "sc-memory/utils/sc_log.hpp"
+
+#include "keynodes/sections_keynodes.hpp"
+
+#include "get_subdivision_agent.hpp"
+#include "constants/sections_aliases.hpp"
+
+using namespace utils;
+
+namespace sectionsModule
+{
+SC_AGENT_IMPLEMENTATION(GetSubdivisionAgent)
+{
+  ScAddr const questionNode = m_memoryCtx.GetEdgeTarget(edgeAddr);
+  if (!CheckActionClass(questionNode))
+    return SC_RESULT_OK;
+
+  SC_LOG_INFO("GetSubdivisionAgent started");
+
+  ScAddr const conceptAddr =
+      IteratorUtils::getAnyByOutRelation(&m_memoryCtx, questionNode, scAgentsCommon::CoreKeynodes::rrel_1);
+
+  if (!m_memoryCtx.IsElement(conceptAddr))
+  {
+    SC_LOG_ERROR("GetSubdivisionAgent: concept node not found.");
+    utils::AgentUtils::finishAgentWork(&m_memoryCtx, questionNode, false);
+    return SC_RESULT_ERROR_INVALID_PARAMS;
+  }
+
+  ScAddrVector subdivision = GetSubdivision(conceptAddr, SectionsKeynodes::nrel_subdividing);
+  if (subdivision.size() == 2)
```

Comment:
it's not obvious why 2

it's looks a bit like magic number until u check GetSubdivision method itself

Reply:
also perhaps we can choose the correct relation depending on conceptAddr class?